x-pack/plugins/infra/public/components/asset_details/tabs/metadata/add_pin_to_row.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Dispatch } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import type { Field } from './utils';
+
+interface AddMetadataPinToRowProps {
+  fieldName: Field['name'];
+  pinnedItems: Array<Field['name']>;
+  setPinnedItems: Dispatch<React.SetStateAction<Array<Field['name']> | undefined>>;
+}
+
+const PIN_FIELD = i18n.translate('xpack.infra.metadataEmbeddable.pinField', {
+  defaultMessage: 'Pin Field',
+});
+
+export const AddMetadataPinToRow = ({
+  fieldName,
+  pinnedItems,
+  setPinnedItems,
+}: AddMetadataPinToRowProps) => {
+  const handleAddPin = (pin: Field['name']) => {
+    setPinnedItems([...pinnedItems, pin]);
+  };
+
+  const handleRemovePin = (pin: Field['name']) => {
+    if (pinnedItems && pinnedItems.includes(pin)) {
+      setPinnedItems((pinnedItems ?? []).filter((pinName: string) => pin !== pinName));
+    }
+  };
```

Comment:
These handlers are receiving `pin` as parameter. When they are called, we basically pass `fieldName`. I think we could do this instead

```suggestion
  const handleAddPin = () => {
    setPinnedItems([...pinnedItems, fieldName]);
  };

  const handleRemovePin = () => {
    if (pinnedItems && pinnedItems.includes(fieldName)) {
      setPinnedItems((pinnedItems ?? []).filter((pinName: string) => fieldName !== pinName));
    }
  };
```

Reply:
Good point, thanks! I change those 👍